molgenis-data-rest/src/main/java/org/molgenis/data/rest/v2/RestControllerV2.java

```diff
@@ -205,7 +207,14 @@ public void deleteEntity(@PathVariable("entityName") String entityName, @PathVar
 		EntityType entityType = dataService.getEntityType(entityName);
 		Object id = getTypedValue(untypedId, entityType.getIdAttribute());
 
-		dataService.deleteById(entityName, id);
+		if (ATTRIBUTE_META_DATA.equals(entityName))
```

Comment:
Metadata == data, the REST controller shouldn't make any entity specific exceptions.